src/dialect/mod.rs

```diff
@@ -638,6 +638,11 @@ pub trait Dialect: Debug + Any {
         false
     }
 
+    /// Returns true if the dialect supports the `UNLISTEN` statement
+    fn supports_unlisten(&self) -> bool {
+        false
+    }
```

Comment:
Oh actually I was initially wondering if it makes sense to reuse the existing `supports_listen()` flag for this but then also realised we could introduce a `dialect.supports_listen_notify()` to mark the feature itself, replacing `dialect.supports_listen()`, `dialect.supports_notify()` and `dialect.supports_unlisten()`. Thinking since it would be expected for example that a dialect supporting listen should also support notify?

Reply:
Yes, although this part of the syntax might be closely related, I think it's better to separate them as independent features. This is because `listen`, `unlisten` and `notify` might not necessarily be supported together. On this issue, perhaps we could get more input from others to consider it more comprehensively. What do you think? @iffyio

Reply:
@iffyio You make a good point. Currently, there are no dialects that support LISTEN but not NOTIFY, so let's go ahead and combine them for now.Have Done!  :)